### ISSUE TITLE
Batch System localized strings overridden on WPEngine

### ISSUE THREAD

**Issue by mnelson4** [eventespresso/event-espresso-core#895]
## Issue Overview

On https://github.com/eventespresso/event-espresso-core/pull/884, @lorenzocaum found that when there was an error during the batch job (to generate a registration CSV file) instead of showing the EE batch job error message ("An error has occurred and the job has been stopped. Please refresh the page to continue.") it showed "null."

I investigated a little and found it was because the javascript `eei18n` didn't have the property `error_message` like it was specified to have in `EED_Batch::_enqueue_batch_job_scripts_and_styles_and_start_job()`. Instead, it has a property `registration_expiration_notice` that reported some error about a session expiring, which was weird. I assume it's because WPEngine is doing something funny with the session, which is causing that registration expiration notice to be added to `eei18n` which is overwritting what EE had for it.

I think the way the batch system is localizing that variable is probably wrong. I should probably be adding that `error_message` onto `EE_Registry::$i18n_js_strings` and calling `EE_Registry::localize_i18n_js_strings()`. Or I should be giving the variable a more globally unique name like `eebatchi18n` or something. @tn3rb will probably know better.

## Bug report or feature request?

* [x] Bug
* [ ] Feature
* [ ] Neither

## Environment Data:

Version of EE: 
4.9.76
Version of WordPress:4.9.9

PHP Version:  7.0

Browser used:  Firefox Quantum

## Steps to Reproduce (for bugs)

* Go to generate a Registration CSV report
* During the file generation, manually edit `\core\libraries\batch\JobHandlers\RegistrationsReport.php`'s `continue_job()`, and add `$foo->bar();` right after the opening `{`
* You'll see an error during the batch generation, which is expected. But if you're on Lorenzo's WPEngine server, it will just say "Internal Server Error. null" instead of the more descriptive 

## Expected Behaviour

The batch generation should shown an error message saying "An error has occurred and the job has been stopped. Please refresh the page to continue."

## Current Behaviour

Instead it says "null"

## Related Information:

Related discussion: https://github.com/eventespresso/event-espresso-core/pull/884#pullrequestreview-192868695

**Comment by mnelson4** [comment]
Passing to @joshfeck for triage. I'm not sure if there are existing known issues with WPEngine. To my knowledge, this issue only comes up if there is an error during a batch job on WPEngine. I don't have an estimate on debug time, but I think it might be a quick one after getting some feedback on it from Brent. Seeing how I don't know of any customers reporting this, it might not be too high priority.

**Comment by tn3rb** [comment]
Ya @mnelson4's assessment is pretty spot on.

What EED_Batch is currently doing:

```php
wp_localize_script(
    'batch_runner',
    'eei18n',
    array(
        'ajax_url'      => WP_AJAX_URL,
        'is_admin'      => (bool) is_admin(),
        'error_message' => esc_html__(
            'An error occurred and the job has been stopped. Please refresh the page to try again.',
            'event_espresso'
        ),
    )
);
```

is incorrect because it is basically trying to handle setting the `eei18n` variable, which is already being done elsewhere. The `error_message` var is likely null because that is getting overwritten.

> I should probably be adding that `error_message` onto `EE_Registry::$i18n_js_strings`

correct. IF you want that var to be part of the `eei18n` object then you can simply add it to that static property any time before it is rendered to the DOM, like so:

```php
EE_Registry::$i18n_js_strings['eeBatchJobError'] =  esc_html__(
    'An error occurred and the job has been stopped. Please refresh the page to try again.',
    'event_espresso'
);
```

>  and calling `EE_Registry::localize_i18n_js_strings()`

you don't even need to do that as long as your scripts are loading in the footer via `admin_enqueue_scripts` after priority 2, because thoat object is localized by `\EventEspresso\core\domain\services\assets\CoreAssetManager::loadCoreJs()`

> Or I should be giving the variable a more globally unique name like `eebatchi18n` or something.

yes that is highly recommended and always a good best practice to namespace things accordingly

**Comment by joshfeck** [comment]
Alright let's get this fixed.

**Comment by joshfeck** [comment]
Fixed in #1060
